plugins/discourse-calendar/spec/models/discourse_post_event/event_spec.rb

```diff
@@ -673,12 +673,38 @@
           )
         end
 
-        it "returns the next occurrence within the recurrence period" do
-          expect(next_date).not_to be_blank
-          expect(next_date).to be_an(Array)
-          expect(next_date.length).to eq(2)
-          expect(next_date[0]).to eq(Time.utc(2020, 4, 25, 13, 0, 0))
-          expect(next_date[1]).to eq(Time.utc(2020, 4, 25, 14, 0, 0))
+        context "when the event has timezone set" do
+          it "returns the next occurrence in UTC timezone" do
+            expect(next_date).not_to be_blank
+            expect(next_date).to be_an(Array)
+            expect(next_date.length).to eq(2)
+            expect(next_date[0]).to eq(Time.utc(2020, 4, 25, 13, 0, 0))
+            expect(next_date[1]).to eq(Time.utc(2020, 4, 25, 14, 0, 0))
+          end
+
+          it "returns the next occurrence in UTC+3 timezone" do
+            event.timezone = "Europe/Kyiv"
+            expect(next_date).not_to be_blank
+            expect(next_date).to be_an(Array)
+            expect(next_date.length).to eq(2)
+            expect(next_date[0]).to eq(Time.utc(2020, 4, 25, 13, 0, 0))
+            expect(next_date[1]).to eq(Time.utc(2020, 4, 25, 14, 0, 0))
+            expect(next_date[0]).to eq(Time.parse("2020-04-25 16:00:00 +0300"))
+            expect(next_date[1]).to eq(Time.parse("2020-04-25 17:00:00 +0300"))
+          end
+
+          it "returns the next occurrence in UTC-7 timezone" do
```

Comment:
with any UTC-3 or further west (-4,-5,etc.) there is no next event date generated - need to figure out, why not